sycl/doc/extensions/experimental/sycl_ext_matrix/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -0,0 +1,926 @@
+of the link:sycl_ext_intel_matrix.asciidoc[sycl_ext_intel_matrix]
+
+Besides the `Group` and the `joint_matrix` arguments,
+`joint_matrix_apply` takes a C++ Callable object which is invoked once
+for each element of the matrix. This callable object must be invocable
+with a single parameter of type `T&`. Commonly, applications pass a
+lambda expression.
+
+```c++
+namespace sycl::ext::oneapi::experimental::matrix {
+
+template<typename Group, typename T, use Use, size_t Rows, size_t Cols,
+  layout Layout, typename F>
+void joint_matrix_apply(Group g, joint_matrix<Group, T, Use, Rows, Cols,
+  Layout>& C, F&& func);
+
+} // namespace sycl::ext::oneapi::experimental::matrix
+```
+
+In the following example, every element of the matrix `C` is
+multiplied by `alpha`. Then, an activation function, `relu` in this
+example, is applied on each of the elements of `C`.
+
+```c++
+joint_matrix_apply(sg, C, [=](T &x) {
+    x *= alpha;
+    relu(x);
+});
+```
+
+=== Support for the TF32 Data Type
+Some devices support the TF32 floating point type for matrix
+elements. This type has a 19 bit format with one sign bit, 8 exponent
+bits (offering the same range as float), and 10 mantissa bits
+(offering the same precision as sycl::half). Use of this type can
+accelerate the joint_matrix_mad operation by reducing its
+precision. In order to declare a `joint_matrix` object with this
+element type, use `matrix::precision::tf32` in place of the `T`
+template parameter.
+
+```c++
+namespace sycl::ext::oneapi::experimental::matrix::precision {
+
+class tf32;
+
+} // namespace sycl::ext::oneapi::experimental::matrix::precision
+```
+
+For example:
+
+```c++
+joint_matrix<sub_group, precision::tf32, use::a, tM, tK,
+             layout::row_major> tA;
+```
+
+Whenever the application loads, stores, fills, or accesses the
+elements of a TF32 matrix, the application sees the elements as
+float. There are special overloads of these functions for TF32 for
+this purpose.
+
+==== TF32 load
+These overloads of `joint_matrix_load` load float values into a TF32
+matrix. It is unspecified whether the implementation loads all 32 bits
+into the joint matrix or if it only loads the relevant 19 bits.
+
+```c++
+namespace sycl::ext::oneapi::experimental::matrix {
+
+template <typename Group, size_t Rows, size_t Cols,
+          access::address_space Space, access::decorated IsDecorated>
+void joint_matrix_load(Group g,
+    joint_matrix<Group, precision::tf32, use::accumulator, Rows, Cols,
+    layout::dynamic> &res,
+    multi_ptr<const float, Space, IsDecorated> src, size_t stride, layout Layout);
+
+// Only available when Layout != layout::dynamic
+template <typename Group, size_t Rows, size_t Cols,
+          use Use, layout Layout,
+          access::address_space Space, access::decorated IsDecorated>
+void joint_matrix_load(Group g,
+    joint_matrix<Group, precision::tf32, Use, Rows, Cols, Layout> &res,
+    multi_ptr<const float, Space, IsDecorated> src, size_t stride);
```

Comment:
I saw that you changed these to `const float` in bf8e00c.  Did you check that this works if the user passes a `multi_ptr` with non-const `float`?  I forget if there is an implicit conversion from non-const `T` to `const T`.